runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -262,41 +272,190 @@ public long getConflictSetAndCommit(ConflictSetInfo conflictSet) {
+
+        // This map contains the maximum address of the streams we have
+        // verified to not have any true conflicts so far.
+        final Map<UUID, Long> verifiedStreams = new HashMap<>();
+
+        // We resolve conflicts until we have a -true- conflict.
+        // This might involve having the sequencer reject our
+        // request to commit multiple times.
+        while (currentException.getAbortCause() == AbortCause.CONFLICT) {
+            final UUID conflictStream = currentException.getConflictStream();
+            final long currentAddress = currentException.getConflictAddress();
+            final TransactionAbortedException thisException = currentException;
+
+            // Get the proxy, which should be available either in the write set
+            // or read set. We need the proxy to generate the conflict objects
+            // from the SMR entry.
+            ICorfuSMRProxyInternal proxy;
+            Optional<ICorfuSMRProxyInternal> modifyProxy = getModifiedProxies().stream()
+                    .filter(p -> p.getStreamID().equals(conflictStream))
+                    .findFirst();
+            if (modifyProxy.isPresent()) {
+                proxy = modifyProxy.get();
+            } else {
+                modifyProxy = getReadSetInfo().getProxy(conflictStream);
+                if (!modifyProxy.isPresent()) {
+                    modifyProxy = getWriteSetInfo().getProxy(conflictStream);
+                    if (!modifyProxy.isPresent()) {
+                        log.warn("doPreciseCommit[{}]: precise conflict resolution requested "
+                                + "but proxy not found, aborting", this);
+                    }
+                    throw currentException;
+                }
+                proxy = modifyProxy.get();
+            }
+
+            // Otherwise starting from the snapshot address to the conflict
+            // address (following backpointers if possible), check if there
+            // is any conflict
+            log.debug("doPreciseCommit[{}]: conflictStream {} searching {} to {}",
+                    this,
+                    Utils.toReadableId(conflictStream),
+                    getSnapshotTimestamp() + 1,
+                    currentAddress);
+            // Generate a view over the stream that caused the conflict
+            IStreamView stream =
+                    builder.runtime.getStreamsView().get(conflictStream);
+            try {
+                ISMRStream smrStream = new StreamViewSMRAdapter(builder.runtime, stream);
+                // Start the stream right after the snapshot.
+                smrStream.seek(getSnapshotTimestamp() + 1);
+                // Iterate over the stream, manually checking each conflict object.
+                smrStream.streamUpTo(currentAddress)
+                        .forEach(x -> {
+                            Object[] conflicts =
+                                    proxy.getConflictFromEntry(x.getSMRMethod(),
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "getSMRMethod"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)